src/mlpack/tests/lars_test.cpp

```diff
@@ -421,4 +421,36 @@ BOOST_AUTO_TEST_CASE(LARSTestComputeError)
   BOOST_REQUIRE_EQUAL(cost == train1, true);
 }
 
+BOOST_AUTO_TEST_CASE(LARSCopyConstructorTest)
+{
+  arma::mat features, Y;
+  arma::rowvec targets;
+  data::Load("lars_dependent_x.csv", features);
+  data::Load("lars_dependent_y.csv", Y);
+  targets = Y.row(0);
+  mlpack::regression::LARS* glm1 = new mlpack::regression::LARS(false, .1, .1);
+  arma::rowvec predictions, predictionsFromCopiedModel;
+  std::vector<mlpack::regression::LARS> models;
+  glm1->Train(features, targets);
+  glm1->Predict(features, predictions);
+  models.emplace_back(*glm1);
+  delete glm1; // Free LARS internal memory.
+  models[0].Predict(features, predictionsFromCopiedModel);
+  CheckMatrices(predictions, predictionsFromCopiedModel);
+  // Check if we can train the model again.
+  BOOST_REQUIRE_NO_THROW(models[0].Train(features, targets));
```

Comment:
It might be nice to add a little whitespace and commenting, i.e., split this code up into a few logical blocks, for the sake of readability.